delta-lake/common/src/main/databricks/scala/com/nvidia/spark/rapids/delta/RapidsDeltaUtils.scala

```diff
@@ -99,4 +101,35 @@ object RapidsDeltaUtils {
     spark.sessionState.conf
       .getConfString("deletionVectors.disableTightBoundOnFileCreationForDevOnly", "false")
       .toBoolean
+
+  /**
+   * Helper method to check if low shuffle merge could be enabled.
+   */
+  def shouldUseLowShuffleMerge(conf: RapidsConf, targetTableSchema: StructType): Boolean = {
+    if (!conf.isDeltaLowShuffleMergeEnabled) {
+      false
+    } else {
+      if (!ScanExecShims.isGpuFileSourceScanExecRuleEnabled(conf)) {
+        logWarning(
+          s"""Delta lake low shuffle merge can't be enabled because GpuFileSourceScanExec is
+             |disabled""".stripMargin)
+        false
+      } else {
+        // Check if all target file types supported by [[GpuFileSourceScanExec]]
+        val unsupportedFields = targetTableSchema.fields
+          .filterNot(f => ScanExecShims.supportByGpuFileSourceScanExec(f.dataType))
+
+        if (unsupportedFields.nonEmpty) {
+          logWarning(
+            s"""Delta lake low shuffle merge can't be enabled because following fields in
+               |target table could be supported by GpuFileSourceScanExec:
+               |${unsupportedFields.mkString("Array(", ", ", ")")}"""
+              .stripMargin)
+          false
+        } else {
```

Comment:
Honestly I'm not thrilled with the extra new exec node, and it's dangerous as I mentioned here to have it, because it makes blind assumptions about its input that may not be correct if the upstream stuff didn't do it's job as expected.

I think it would be safer and cleaner if GpuFileSourceScanExec directly advertised it's output as being hash partitioned by the file path when it's in this special mode.  In other words, put the logic of the new exec in GpuFileSourceScanExec.  Then we're safely covered if the scan falls back to the CPU, because Spark itself will inject the necessary shuffle for the join rather than performing an invalid join and producing bad output.  It also removes the extra exec from the plan which may confuse users as to what its doing (as it has no CPU correlation).

Reply:
Ah right, I was thinking these would be injected between GpuOverrides and GpuOverrideTransitions, but that's only columnar transitions not shuffle exchanges.